scripts/plugins/support-pssl.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+#
+# Copyright (c) Citrix Systems 2017. All rights reserved.
+
+set -e
+
+# Open/Close port 6640 when pssl is used in ovs
+
+
+OP="$1"
+CHAIN="RH-Firewall-1-INPUT"
+POSITION=$(iptables -L --line-n -n | grep -A 11 "Chain $CHAIN" | grep "443" | cut -d " " -f1)
```

Comment:
I'm not entirely happy about searching for "443" to decide where to insert the rule, either. What if there's a "1443" somewhere, or if it's for a udp rule or something?

But how did you decide that just before the https port was a good place to insert the new rule anyway? What would happen if you always inserted the new rule as rule 1?

But any approach like this might stop working in the future if we have a different default list of rules in the chain.

How about making a new chain instead, a chain which is totally managed by this script, and can be empty? Then make a permanent change to the configuration so that your new chain is used as well as the existing ones.